src/app/details/_smart-components/recipe-form/recipe-form.component.ts

```diff
@@ -30,19 +28,19 @@ export class RecipeFormComponent
   maxNameLength: number = 80;
   maxDescriptionLength: number = 255;
 
-  constructor(
-    private _fb: FormBuilder,
-    private _cdr: ChangeDetectorRef,
-    private _router: Router
-  ) {
+  constructor(private _fb: FormBuilder, private _cdr: ChangeDetectorRef) {
     super();
   }
 
-  override ngOnInit(): void {
-    super.ngOnInit();
+  ngOnInit(): void {
     this.recipeData$?.pipe(takeUntil(this.destroy$)).subscribe({
       next: (recipe) => {
         this.recipeData = recipe;
+        if (this.recipeData) {
+          this.editMode = true;
+        } else {
+          this.editMode = false;
+        }
```

Comment:
rozumiem że jak jest undefined albo null to ma być false. można to zrobić czyściej używając podwójnego wykrzyknika, czyli this.editMode = !!this.recipeData